app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -223,6 +241,57 @@ export default class PendingTransactionTracker extends EventEmitter {
           baseFeePerGas,
           blockTimestamp,
         );
+        // Adds tokens to state on confirmation of transaction
+        // if the sender is a recipient of the token.
+        const txSender = transactionReceipt?.from;
+        const tokenTransfers = getTokenTransfersFromTxReceipt(
+          transactionReceipt,
+        ).filter(
+          (transfer) => transfer.to.toLowerCase() === txSender.toLowerCase(),
+        );
+        const deduplicatedTransferContractsAddresses = [
+          ...new Set(
+            tokenTransfers.map((transfer) =>
+              transfer.contractAddress.toLowerCase(),
+            ),
+          ),
+        ];
+
+        const tokens = await Promise.all(
+          deduplicatedTransferContractsAddresses.map((contractAddress) =>
+            this.getTokenStandardAndDetails(contractAddress, txSender),
+          ),
+        );
+
+        const receivedTokens = tokens
+          .filter((token) => !isEmpty(token))
+          .reduce((acc, token) => {
+            acc[token.contractAddress.toLowerCase()] = {
+              ...token,
+              image: formatIconUrlWithProxy(txMeta.chainId, token.address),
+            };
+            return acc;
+          }, {});
```

Comment:
there is no need to use filter then reduce , you can just add if statement inside your reduce callback funcion.
e.g:
`const receivedTokens = tokens.reduce((acc, token) => {
  if (!isEmpty(token)) {
    const { contractAddress, ...restToken } = token;
    acc[contractAddress.toLowerCase()] = {
      ...restToken,
      image: formatIconUrlWithProxy(txMeta.chainId, token.address),
    };
  }
  return acc;
}, {});`